docs/json_schemas/tls_certificates/v1/provider.json

```diff
@@ -0,0 +1,122 @@
+{
+  "$defs": {
+    "BaseModel": {
+      "properties": {},
+      "title": "BaseModel",
+      "type": "object"
+    },
+    "Certificate": {
+      "description": "Certificate model.",
+      "properties": {
+        "ca": {
+          "description": "Whether the certificate is a CA.",
+          "title": "Ca",
+          "type": "string"
+        },
+        "certificate_signing_request": {
+          "description": "Certificate signing request.",
+          "title": "Certificate Signing Request",
+          "type": "string"
+        },
+        "certificate": {
+          "description": "Certificate.",
+          "title": "Certificate",
+          "type": "string"
+        },
+        "chain": {
+          "anyOf": [
+            {
+              "items": {
+                "type": "string"
+              },
+              "type": "array"
+            },
+            {
+              "type": "null"
+            }
+          ],
+          "description": "List of certificates in the chain.",
+          "title": "Chain"
+        },
+        "recommended_expiry_notification_time": {
+          "anyOf": [
+            {
+              "type": "integer"
+            },
+            {
+              "type": "null"
+            }
+          ],
+          "description": "Recommended expiry notification time in seconds.",
+          "title": "Recommended Expiry Notification Time"
+        },
+        "revoked": {
+          "anyOf": [
+            {
+              "type": "boolean"
+            },
+            {
+              "type": "null"
+            }
+          ],
+          "description": "Whether the certificate is revoked.",
+          "title": "Revoked"
+        }
+      },
+      "required": [
+        "ca",
+        "certificate_signing_request",
+        "certificate",
+        "chain",
```

Comment:
These are marked optional in the `schema.py` isn't there a contradiction?

Reply:
I think this is an unfortunate side-effect of pydantic/json schema's implementation of optional.

The keys are required, but the value can be null.